### PR TITLE
Rubocop refactoring and setup wiil_paginate initializer and profile page

### PR DESCRIPTION
**Pull request by marderer**
Setup will_paginate initizlizer, install group rubocop and refactoring and fix bug with edit profile page

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -2,35 +2,192 @@ AllCops:
...
+Style/MethodDefParentheses:
+  EnforcedStyle: require_parentheses
+
+Style/RedundantReturn:
+  AllowMultipleReturnValues: false
+
+Style/StringLiterals:
+  EnforcedStyle: single_quotes
+  ConsistentQuotesInMultiline: true
+
```
Comment (AleksSenkou) [inline .rubocop.yml]: 👍

File: Gemfile [hunk 2]
```diff
@@ -1,67 +1,47 @@
...
-  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
   gem 'spring-watcher-listen', '~> 2.0.0'
-  gem "letter_opener"
+  gem 'web-console', '>= 3.3.0'
 end
 
-# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
-gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
```
Comment (AleksSenkou) [inline Gemfile]: 👍

File: .rubocop.yml (hunk 1 above)
Comment (AleksSenkou) [reply .rubocop.yml]: @marderer the only thing: useless empty line at the end of the file

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -5,9 +5,10 @@ class ApplicationController < ActionController::Base
...
   def configure_permitted_parameters
-    devise_parameter_sanitizer.permit(:sign_up, keys: %i[username email password password_confirmation remember_me
-                                                         avatar avatar_cache remove_avatar])
-    devise_parameter_sanitizer.permit(:account_update, keys: %i[username email password password_confirmation
-                                                                current_password avatar avatar_cache remove_avatar])
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[username email password password_confirmation
+                                                         remember_me avatar avatar_cache remove_avatar])
+    devise_parameter_sanitizer.permit(:account_update, keys: %i[username email password
+                                                                password_confirmation current_password
+                                                                avatar avatar_cache remove_avatar])
```
Comment (AleksSenkou) [inline app/controllers/application_controller.rb]: @marderer what do you think about giving a breath to `configure_permitted_parameters` method? 
```ruby
def configure_permitted_parameters
  devise_parameter_sanitizer.permit(:sign_up, keys: shared_permitted_keys.push(:remember_me))
  devise_parameter_sanitizer.permit(:account_update, keys: shared_permitted_keys)
end

private

def shared_permitted_keys
  %i[
    username
    email 
    password 
    password_confirmation 
    remember_me 
    avatar 
    avatar_cache 
    remove_avatar
  ]
end
```

File: app/models/user.rb [hunk 4]
```diff
@@ -1,9 +1,8 @@
...
-  has_many :transactions
+  has_many :transactions, dependent: :destroy
 
   mount_uploader :avatar, AvatarUploader
 
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable, :omniauthable, :confirmable
-  devise :omniauthable, omniauth_providers: %i[google_oauth2 github]
+  devise :database_authenticatable, :registerable, :recoverable, :rememberable, :validatable
+  devise :confirmable, :omniauthable, omniauth_providers: %i[google_oauth2 github]
```
Comment (AleksSenkou) [inline app/models/user.rb]: @marderer why did you split `devise` options to two lines and moved `:confirmable` to the second line
As for we - `devise :omniauthable, omniauth_providers: %i[google_oauth2 github]` looks great

File: app/views/devise/registrations/edit.html.slim [hunk 5]
```diff
@@ -34,12 +33,12 @@ h2
...
   .form-group
     = f.label :current_password
-    i (we need your current password to confirm your changes)
-    br/
-    = f.password_field :current_password, autocomplete: "off"
-  .actions
-    = f.submit t('update')
-h3 Cancel my account
+    i
+      | (we need your current password to confirm your changes)
```
Comment (AleksSenkou) [inline app/views/devise/registrations/edit.html.slim]: locales

File: bin/update [hunk 6]
```diff
@@ -1,18 +1,15 @@
...
 APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
 
 def system!(*args)
   system(*args) || abort("\n== Command #{args} failed ==")
 end
 
 chdir APP_ROOT do
-  # This script is a way to update your development environment automatically.
-  # Add necessary update steps to this file.
+  include FileUtils
```
Comment (AleksSenkou) [inline bin/update]: @marderer is it rubocop who forced you to move `include FileUtils` from the beginning of the file? If yes - let's disable this cop and move `include FileUtils` back at the start of the file. If no - describe why have you done it?

File: reek.yml [hunk 7]
```diff
@@ -9,4 +9,8 @@ InstanceVariableAssumption:
 UncommunicativeMethodName:
   enabled: false
```
Comment (AleksSenkou) [inline reek.yml]: @marderer did you try to enable `UncommunicativeMethodName`?

File: bin/update (hunk 6 above)
Comment (marderer) [reply bin/update]: It is rubocop

File: reek.yml (hunk 7 above)
Comment (marderer) [reply reek.yml]: @AleksSenkou yes, I try, but I don't know what do with `google_oauth2`

Comment (AleksSenkou) [reply reek.yml]: @marderer try `config.omniauth :google_oauth2, "GOOGLE_CLIENT_ID", "GOOGLE_CLIENT_SECRET", { name: "google" }`

Comment (AleksSenkou) [reply reek.yml]: `UncommunicativeMethodName` is super important metric

File: app/views/devise/passwords/edit.html.slim [hunk 8]
```diff
@@ -6,12 +6,13 @@ h2= t('сhange_password')
     = f.label :password, "New password"
     - if @minimum_password_length
       em
-        | (#{@minimum_password_length} characters minimum)
+        | (#{@minimum_password_length}
```
Comment (AleksSenkou) [inline app/views/devise/passwords/edit.html.slim]: @marderer can't we go with:
```ruby
| (
= @minimum_password_length
```
?

File: app/views/devise/registrations/edit.html.slim [hunk 9]
```diff
@@ -18,28 +17,29 @@ h2
     = f.email_field :email, placeholder: 'Email', class: 'form-control'
     - if devise_mapping.confirmable? && resource.pending_reconfirmation?
       div
-        | Currently waiting confirmation for: #{resource.unconfirmed_email}
+        = t('currently_waiting_confirmation_for')
+        | : #{resource.unconfirmed_email}
```
Comment (AleksSenkou) [inline app/views/devise/registrations/edit.html.slim]: the same thing here

File: app/views/devise/registrations/new.html.slim [hunk 10]
```diff
@@ -9,7 +9,9 @@ h2= t('sign_up')
     = f.label :password
     - if @minimum_password_length
       em
-        | (#{@minimum_password_length} characters minimum)
+        | (#{@minimum_password_length}
```
Comment (AleksSenkou) [inline app/views/devise/registrations/new.html.slim]: and there

**Pull request merged**